applications/StructuralMechanicsApplication/tests/cpp_tests/test_truss_constitutive_law.cpp

```diff
@@ -0,0 +1,59 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:         BSD License
+//                   license: StructuralMechanicsApplication/license.txt
+//
+//  Main authors:    Richard Faasse
+//
+
+#include "structural_mechanics_fast_suite.h"
+#include "custom_constitutive/truss_constitutive_law.h"
+
+namespace Kratos::Testing
+{
+
+KRATOS_TEST_CASE_IN_SUITE(TrussConstitutiveLaw_CalculatesLinearElasticStress, KratosStructuralMechanicsFastSuite) {
+    TrussConstitutiveLaw law;
+    ConstitutiveLaw::Parameters parameters;
+    Vector temp_strain = ScalarVector(1, 0.005);
+    Vector temp_stress = ZeroVector(1);
+    parameters.SetStrainVector(temp_strain);
+    parameters.SetStressVector(temp_stress);
+
+    Properties properties;
+    properties[YOUNG_MODULUS] = 1e6;
+    parameters.SetMaterialProperties(properties);
+    law.CalculateMaterialResponsePK2(parameters);
+
+    constexpr double expected_stress = 5000; // = Strain * Young's Modulus
+    KRATOS_EXPECT_EQ(expected_stress, parameters.GetStressVector()[0]);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(TrussConstitutiveLaw_CalculatesLinearElasticStress_WithInitialStress, KratosStructuralMechanicsFastSuite) {
+    TrussConstitutiveLaw law;
+    Vector temp_strain = ScalarVector(1, 0.005);
+    Vector temp_stress = ZeroVector(1);
+
+    ConstitutiveLaw::Parameters parameters;
+    parameters.SetStrainVector(temp_strain);
+    parameters.SetStressVector(temp_stress);
+
+    Properties properties;
+    properties[YOUNG_MODULUS] = 1e6;
+    parameters.SetMaterialProperties(properties);
+
+    Vector initial_strain = ZeroVector(1);
+    Vector initial_stress = ScalarVector(1, 5000);
+    InitialState::Pointer p_initial_state = Kratos::make_intrusive<InitialState>(initial_strain, initial_stress);
```

Comment:
You may also consider to default-construct an `InitialState` instance, and then call `SetInitialStressVector` to set the initial stress vector only. This saves you from defining a "dummy" initial strain vector.

Reply:
Good point, done